### DISCUSSION TITLE
Watch and Shared Informer

### DISCUSSION THREAD

**Discussion by victpf** [kubernetes-sigs/kubebuilder#2453] [Support] [answered]
Under what kind of circumstances should we use Watch rather than SharedInformer, and vise versa?
I know that SharedInformer create/main objects' CRUD in cache which is more efficient and contain full kinds of error handling logics.
However, I see that people would use Watch within SetupWIthManager() function when developing operators.
Can we use shared informers here in SetupWIthManager? and the reason why?
I want to know if we should always use SharedInformer? Or do we have any guidelines here about how to determine which one to use?

**Comment by camilamacedo86** [comment, marked as answer]
The information that you are looking for seems to be described in the doc: https://book.kubebuilder.io/reference/watching-resources

If that does not answer your question please feel free to raise an issue.
